encoding_specification.md

```diff
@@ -0,0 +1,93 @@
+---
+title: Encoding Specification
+---
+
+## SOMA Encoding Version 2.0.0
+
+SOMA encoding version 2.0.0 introduces per type versioning. T
+
+### `SOMACollection`
+
+#### `SOMACollection` Type Encoding Version 1.0
+
+| Metadata key               | Datatype           | Value            | Description                         | Required |
+| :------------------------- | :----------------- | :--------------- | :---------------------------------- | :------- |
+| soma_datatype              | TILEDB_STRING_UTF8 | "SOMACollection" | SOMA type name                      | required |
+| soma_encoding_version      | TILEDB_STRING_UTF8 | "2.0.0"          | general SOMA encoding version       | required |
+| soma_type_encoding_version | TILEDB_STRING_UTF8 | "1.0"            | type-specific SOMA encoding version | required |
```

Comment:
This proposal adds in a per-type encoding version. This will allow us to update the version of individual elements of the dataset without updating (and causing forward-compatibility issues) for all SOMA datatypes.